bob-apps/bin/bobt.rs

```diff
@@ -0,0 +1,434 @@
+        self.delete_count += 1;
+        let res = self.get(key).await;
+        match res {
+            Ok(new_size) => Err(format!("{} => {}", size, new_size)),
+            Err(_) => Ok(size),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct Settings {
+    count: u64,
+    start_id: u64,
+    end_id: u64,
+    max_size: usize,
+    key_size: usize,
+    uri: Uri,
+    api_uri: Uri,
+}
+
+impl Settings {
+    fn new() -> Self {
+        let matches = get_matches();
+        Self {
+            count: Self::get_count(&matches),
+            uri: Self::get_uri(&matches),
+            start_id: Self::get_start_id(&matches),
+            end_id: Self::get_end_id(&matches),
+            max_size: Self::get_max_size(&matches),
+            key_size: Self::get_key_size(&matches),
+            api_uri: Self::get_api_uri(&matches),
+        }
+    }
+
+    fn get_count(matches: &ArgMatches) -> u64 {
+        matches
+            .value_of(COUNT_ARG_NAME)
+            .expect("required")
+            .parse()
+            .expect("should be u64")
+    }
+
+    fn get_max_size(matches: &ArgMatches) -> usize {
+        matches
+            .value_of(MAX_SIZE_ARG_NAME)
+            .expect("has default")
+            .parse()
+            .expect("should be usize")
+    }
+
+    fn get_start_id(matches: &ArgMatches) -> u64 {
+        matches
+            .value_of(START_ID_ARG_NAME)
+            .expect("has default")
+            .parse()
+            .expect("should be u64")
+    }
+
+    fn get_end_id(matches: &ArgMatches) -> u64 {
+        matches
+            .value_of(END_ID_ARG_NAME)
+            .expect("has default")
+            .parse()
+            .expect("should be u64")
+    }
+
+    fn get_uri(matches: &ArgMatches) -> Uri {
+        matches
+            .value_of(URI_ARG_NAME)
+            .expect("has default value")
+            .parse()
+            .expect("wrong format of url")
+    }
+
+    fn get_key_size(matches: &ArgMatches) -> usize {
+        matches
+            .value_of(KEY_SIZE_ARG_NAME)
+            .expect("has default value")
+            .parse()
+            .expect("wrong format of url")
+    }
+
+    fn get_api_uri(matches: &ArgMatches) -> Uri {
+        matches
+            .value_of(API_ADDRESS_ARG_NAME)
+            .expect("has default value")
+            .parse()
+            .expect("wrong format of url")
+    }
+
+    fn convert_key(&self, key: u64) -> String {
+        let mut data = key.to_le_bytes().to_vec();
+        data.resize(self.key_size, 0);
+        let mut result = String::with_capacity(data.len() * 2 + 2);
+        result.push_str("0x");
+        for b in data {
+            result.push_str(&format!("{:X?}", b));
+        }
+        result
+    }
+}
+
+fn get_matches<'a>() -> ArgMatches<'a> {
+    let count_arg = Arg::with_name(COUNT_ARG_NAME)
+        .short("c")
+        .long("count")
+        .takes_value(true)
+        .required(true);
+    let uri_arg = Arg::with_name(URI_ARG_NAME)
```

Comment:
This argument can be removed, and `short("a")` can be used for `api-address` parameter